index.js

```diff
@@ -22,35 +26,57 @@ function initialize(projectId, options) {
     appId: '1:757397537758:web:7dd1645537045fdfcb534f'
   };
 
+  pId = projectId;
+
   firebase.initializeApp(firebaseConfig);
   const database = firebase.database();
 
   currVersion = localStorage.getItem('jellySyncVersion');
+  endpoint = localStorage.getItem('jellysyncEndpoint');
+
+  if (!endpoint) {
+    endpoint = await getEndpoint(projectId);
+  }
 
-  const ref = database.ref(`projects/${projectId}`);
-  connect(ref);
+  dbRef = database.ref(`project/${endpoint}`);
+  connect();
 
-  ref.onDisconnect(() => connect(ref));
+  ref.onDisconnect(() => connect());
 }
 
-function connect(ref) {
-  ref.on('value', snapshot => {
-    const snapshotValue = snapshot.val();
+async function connect() {
+  try {
+    dbRef.on('value', snapshot => {
+      const snapshotValue = snapshot.val();
+
+      if (!currVersion) {
+        localStorage.setItem('jellySyncVersion', snapshotValue.version);
+        currVersion = snapshotValue.version;
 
-    if (!currVersion) {
-      localStorage.setItem('jellySyncVersion', snapshotValue.version);
-      currVersion = snapshotValue.version;
+        return;
+      }
 
-      return;
-    }
+      if (currVersion !== snapshotValue.version) {
+        (snapshotValue.actions || []).forEach(action => actionFunctions[action](snapshotValue));
+
+        localStorage.setItem('jellySyncVersion', snapshotValue.version);
+        currVersion = snapshotValue.version;
+      }
+    });
+  } catch (e) {
+    endpoint = await getEndpoint(pId);
```

Comment:
so does this just loop if it can't connect?

Reply:
actually technically no cause it doesn't call connect, its kinda doing nothing right now

Reply:
what should we do so it doesn't loop?